intelmq/bots/parsers/shadowserver/config.py

```diff
@@ -138,37 +138,44 @@ def convert_float(value):
         return float(value)
 
 
-def convert_hostname_and_url(value, row):
+def convert_http_host_and_url(value, row):
     """
-    URLs are split into hostname and path, we can also guess the protocol here.
-    but only guess if the protocol is in a set of known good values.
+    URLs are split into hostname and path. The column names differ in reports.
+    Compromised-Website: http_host, url
+    Drone: cc_dns, url
+    IPv6-Sinkhole-HTTP-Drone: http_host, http_url
+    Microsoft-Sinkhole: http_host, url
+    Sinkhole-HTTP-Drone: http_host, url
+    With some reports, url/http_url holds only the path, with others the full HTTP request.
     """
-    if row['application'] in ['http', 'https', 'irc']:
-        if row['hostname'] and row['url']:
-            url = row['url'] if row['url'].startswith('/') else '/' + row['url']
-            return row['application'] + '://' + row['hostname'] + url
+    hostname = ""
+    if "cc_dns" in row:
+        if row['cc_dns']:
+            hostname = row['cc_dns']
+    elif "http_host" in row:
+        if row['http_host']:
+            hostname = row['http_host']
+
+    path = ""
+    if "url" in row:
+        if row['url']:
+            path = row['url']
+    elif "http_url" in row:
+        if row['http_url']:
+            path = row['http_url']
 
-        elif row['hostname'] and not row['url']:
-            return row['application'] + '://' + row['hostname']
+    if hostname and path:
+        # remove potential leading/trailing HTTP request information
+        path = re.sub(r'^[^/]*', '', path)
+        path = re.sub(r'\s.*$', '', path)
 
-    return value
+        application = "http"
+        if "application" in row:
+            if row['application'] in ['http', 'https']:
```

Comment:
Otherwise it's an error, right?

Reply:
Most events don't include a value for application at all, so we should use "http" as default to be used if host and url are set.
Some events have application set to "irc" or to strange values like "3.1002" but then there is no url and the value of application should be disregarded.
As "http" is default, this could also be removed from the list in line 174 so the value is only changed if application is set to "https".